File: 3-leetcode/02-medium/js/624-maximum-distance-in-arrays.ts
const maxDistance = (arrays: number[][]): number => {
  let min = arrays[0][0];
  let max = arrays[0][arrays[0].length - 1];
  let result = 0;

  for (let i = 1; i < arrays.length; i++) {
    const length = arrays[i].length;
    result = Math.max(result, arrays[i][length - 1] - min, max - arrays[i][0]);
    min = Math.min(min, arrays[i][0]);
    max = Math.max(max, arrays[i][length - 1]);
  }

  return result;
};

File: 3-leetcode/02-medium/dart/624-maximum-distance-in-arrays.dart
class Solution {
  int maxDistance(List<List<int>> arrays) {
    int _min = arrays[0][0];
    int _max = arrays[0].last;
    int result = 0;
    for (int i = 1; i < arrays.length; i++) {
      result = max(result, arrays[i].last - _min);
      result = max(result, _max - arrays[i][0]);
      _min = min(_min, arrays[i][0]);
      _max = max(_max, arrays[i].last);
    }
    return result;
  }
}
